### PR TITLE
feat(fallback): support custom entities

### LINKED ISSUE #6122: Support custom entities in failure domain
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does this enhancement require public documentation?

- [X] I have added an Acceptance Criteria item for adding and/or adjusting public documentation (if applicable)

### Problem Statement

Custom entities are not supported in failure domains. So they cannot be recovered with cached good config. This may cause stuck in applying configuration to Kong gateway when custom entities have wrong config. 

### Proposed Solution

- Add custom entities to caches of failure domains
- Generate dependencies of custom entities from their schema

### Additional information

Blocked by #6054

### Acceptance Criteria

- [ ] Custom entities can be recovered in failure domains.

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

Implements support for custom entities in the fallback configuration feature by defining its dependencies-resolving function and covering it with tests.

**Which issue this PR fixes**:

Closes #6122.

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: internal/dataplane/fallback/cache_to_graph_test.go [hunk 1]
```diff
@@ -332,6 +333,49 @@ func TestDefaultCacheGraphProvider_CacheToGraph(t *testing.T) {
...
+							Name:  "test-cluster-plugin",
+						},
+					},
+				},
+				testKongPlugin(t, "test-plugin"),
+				testKongClusterPlugin(t, "test-cluster-plugin"),
+			),
+			expectedAdjacencyMap: map[string][]string{
+				"configuration.konghq.com/KongPlugin:test-namespace/test-plugin": {
+					"configuration.konghq.com/KongCustomEntities:test-namespace/test-entity-kong-plugin",
```
Comment (randmonkey) [inline internal/dataplane/fallback/cache_to_graph_test.go:370]: kind here is `KongCustomEntity` not `KongCustomEntities`.

File: internal/dataplane/fallback/graph_dependencies_kong.go [hunk 2]
```diff
@@ -140,3 +141,36 @@ func resolveTCPIngressDependencies(cache store.CacheStores, tcpIngress *kongv1be
...
+// resolveKongCustomEntityDependencies resolves potential dependencies for a KongCustomEntities object:
+// - KongPlugin
+// - KongClusterPlugin.
+func resolveKongCustomEntityDependencies(cache store.CacheStores, obj *kongv1alpha1.KongCustomEntity) []client.Object {
+	if obj.Spec.ParentRef == nil {
+		return nil
+	}
+
+	parentRef := *obj.Spec.ParentRef
+	groupMatches := parentRef.Group != nil && *parentRef.Group == "configuration.konghq.com"
```
Comment (randmonkey) [inline internal/dataplane/fallback/graph_dependencies_kong.go:154]: Could we replace the group `configuration.konghq.com` here to the constant defined in the API packages?

File: internal/dataplane/testdata/golden/fallback-config-kong-custom-entities/in.yaml [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                port:
+                  number: 80
+---
+apiVersion: configuration.konghq.com/v1
+kind: KongPlugin
+metadata:
+  namespace: default
+  name: p1
+  uid: e97ca177-6bd2-425f-b3b1-10a521ac02a0
+plugin: p1
```
Comment (randmonkey) [inline internal/dataplane/testdata/golden/fallback-config-kong-custom-entities/in.yaml:51]: Looks like we cannot use fake (non-exist) plugins and custom entities when we run golden tests agianst Kong gateway.

Comment (czeslavo) [reply internal/dataplane/testdata/golden/fallback-config-kong-custom-entities/in.yaml]: Yes, sorry I missed that. I've changed the tests to use `degraphql` plugin and `degraphql_routes` as custom entities. Also added `-ee` prefix to the golden tests directory name so it's properly treated as an enterprise test case. 

In https://github.com/Kong/kubernetes-ingress-controller/pull/6293 I'm resolving the issue that we had created to enable EE kongintegration tests.

**Review by randmonkey — APPROVED** [review]

**Pull request merged**